File: travel_guide/lib/Model/destination.model.dart

class Destination {

  late String name;
  late String description;
  late String image;

  Destination({required this.name, required this.description, required this.image});
  // add more properties here

  Destination.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    description = json['description'];
    image = json['image'];
  }

}

File: backend/Models/destinationModel.js


import mongoose from 'mongoose';


const destinationSchema = mongoose.Schema({
    name: {
        type: String,
        required: true,
        unique: true
    },
    location: {
        type: String,
        required: true
    },
    description: {
        type: String,
        required: true
    },
    image: {
        type: String,
        required: false,
    },
    rating: {
        type: Number,
        default: 0,
    },
    things_to_do: {
        type: [String],
        required: true,
    },
    map: {
        type: String,
    }
}, {
    timestamps: true
});

const Destination = mongoose.model('Destination', destinationSchema);

export default Destination;